cli/azd/pkg/auth/manager.go

```diff
@@ -160,25 +176,16 @@ func (m *Manager) CredentialForCurrentUser(
 	if currentUser.HomeAccountID != nil {
 		for _, account := range m.publicClient.Accounts() {
 			if account.HomeAccountID == *currentUser.HomeAccountID {
-				if options.TenantID == "" {
-					return newAzdCredential(m.publicClient, &account), nil
-				} else {
-					newAuthority := "https://login.microsoftonline.com/" + options.TenantID
-
-					newOptions := make([]public.Option, 0, len(m.publicClientOptions)+1)
-					newOptions = append(newOptions, m.publicClientOptions...)
-
-					// It is important that this option comes after the saved public client options since it will
-					// override the default authority.
-					newOptions = append(newOptions, public.WithAuthority(newAuthority))
-
-					clientWithNewTenant, err := public.New(cAZD_CLIENT_ID, newOptions...)
-					if err != nil {
-						return nil, err
-					}
-
-					return newAzdCredential(&msalPublicClientAdapter{client: &clientWithNewTenant}, &account), nil
+				realm := account.Realm
+				if options.TenantID != "" {
+					realm = options.TenantID
```

Comment:
We are getting to the point of having many if/else paths which make the code more complex to follow.  Consider breaking this up into multiple methods to avoid the complexity.